src/applications/model/gaming-streaming-server.cc

```diff
@@ -117,52 +119,48 @@ GamingStreamingServer::AddNewTraffic (Ptr<RandomVariableStream> packetSize,
                                       Ptr<RandomVariableStream> interArrivalTime)
 {
   NS_LOG_FUNCTION (this << packetSize << interArrivalTime);
-  m_packetSizeVariables.push_back (packetSize);
-  m_interArrivalTimesVariables.push_back (interArrivalTime);
-  m_sendEvent.push_back (EventId ());
+  TrafficStream newTraffic;
+  newTraffic.m_packetSizeVariable = packetSize;
+  newTraffic.m_interArrivalTimesVariable = interArrivalTime;
+  newTraffic.m_sendEvent = EventId ();
+  m_trafficStreams.push_back (newTraffic);
 }
 
 void
 GamingStreamingServer::Send (int idx)
 {
   NS_LOG_FUNCTION (this);
 
-  uint32_t pktSize = std::max ((int)m_packetSizeVariables[idx]->GetInteger (), 12); //TODO: think about 12
+  uint32_t pktSize = m_trafficStreams[idx].m_packetSizeVariable->GetInteger ();
 
-  SeqTsHeader header;
-  header.SetSeq (m_seq++);
-  NS_ABORT_IF (pktSize < header.GetSerializedSize ());
-  Ptr<Packet> packet = Create<Packet> (pktSize - header.GetSerializedSize ());
-  packet->AddHeader (header);
+  Ptr<Packet> packet = Create<Packet> (pktSize);
+  TimestampTag timestamp;
+  timestamp.SetTimestamp (Simulator::Now ());
+  packet->AddByteTag (timestamp);
   NS_ABORT_IF (packet->GetSize () != pktSize);
 
   m_socket->Send (packet);
```

Comment:
`Socket::Send` returns the number of bytes successfully transmitted if no error, otherwise -1. (https://www.nsnam.org/doxygen/classns3_1_1_socket.html#a3c439c8d005bf0fdd2480c93a0e8056c).
Maybe we should use this information to check if we are sending the entire packet and mostly if we are sending the packet without problems